packages/edit-post/src/components/options-modal/options/enable-custom-fields.js

```diff
@@ -1,47 +1,54 @@
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
 import { withSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import BaseOption from './base';
 
-export class EnableCustomFieldsOption extends Component {
-	constructor( { isChecked } ) {
-		super( ...arguments );
-
-		this.toggleCustomFields = this.toggleCustomFields.bind( this );
-
-		this.state = { isChecked };
-	}
-
-	toggleCustomFields() {
-		// Submit a hidden form which triggers the toggle_custom_fields admin action.
-		// This action will toggle the setting and reload the editor with the meta box
-		// assets included on the page.
-		document.getElementById( 'toggle-custom-fields-form' ).submit();
-
-		// Make it look like something happened while the page reloads.
-		this.setState( { isChecked: ! this.props.isChecked } );
-	}
-
-	render() {
-		const { label } = this.props;
-		const { isChecked } = this.state;
+export function CustomFieldsConfirmation( { nextState } ) {
```

Comment:
Am thinking something like `willEnable` is a more descriptive name for this prop, but it's not a big deal since this component is only used in this file.

Reply:
It took me the longest time to come up with the bad name. `willEnable` is 💯